### PR TITLE
Support complex type.

### PR DESCRIPTION
**Pull request by karataliu**

### CODE REVIEW

File: src/Microsoft.Restier.EntityFramework/Properties/Resources.resx [hunk 1]
```diff
@@ -124,4 +124,7 @@
     <value>Could not find the specified resource.</value>
     <comment>The target entity not found for modification</comment>
   </data>
+  <data name="UnsupportedStructurePropertyType" xml:space="preserve">
```
Comment (congysu) [inline src/Microsoft.Restier.EntityFramework/Properties/Resources.resx]: Unsupported`Structural`PropertyType

File: test/Microsoft.Restier.EntityFramework.Tests/ModelTests.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+{
+    public class ModelBuilderTest
+    {
+        [Fact]
+        public void ComplexTypeShoudWork()
+        {
+            var model = Domain.GetModelAsync(new LibraryDomain().Context).Result;
+            IEnumerable<EdmError> errors;
+            Assert.True(model.Validate(out errors));
+            Assert.Equal(0, errors.Count());
```
Comment (congysu) [inline test/Microsoft.Restier.EntityFramework.Tests/ModelTests.cs]: `Assert.Empty(errors);`

File: test/Microsoft.Restier.EntityFramework.Tests/Models/Library/Address.cs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+﻿namespace Microsoft.Restier.EntityFramework.Tests.Models.Library
```
Comment (congysu) [inline test/Microsoft.Restier.EntityFramework.Tests/Models/Library/Address.cs]: License header, and other new files.

File: test/Microsoft.Restier.EntityFramework.Tests/packages.config [hunk 4]
```diff
@@ -0,0 +1,6 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="EntityFramework" version="6.1.1" targetFramework="net45" />
+  <package id="Microsoft.OData.Edm" version="6.9.0" targetFramework="net45" />
```
Comment (congysu) [inline test/Microsoft.Restier.EntityFramework.Tests/packages.config]: Not for this change: We shall update the ODataLib dependency to a newer version.

File: test/Microsoft.Restier.EntityFramework.Tests/packages.config [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+  <package id="xunit" version="1.9.2" targetFramework="net45" />
```
Comment (congysu) [inline test/Microsoft.Restier.EntityFramework.Tests/packages.config]: Not for this change: We shall move to xUnit 2.0.

File: test/ODataEndToEndTests/Microsoft.Restier.WebApi.Test.Scenario/TrippinE2ETestCases.cs [hunk 6]
```diff
@@ -114,6 +115,71 @@ public void CURDEntity()
         }
 
         [Fact]
+        public void CURDEntityWithComplexType()
+        {
+            this.TestClientContext.MergeOption = MergeOption.OverwriteChanges;
```
Comment (congysu) [inline test/ODataEndToEndTests/Microsoft.Restier.WebApi.Test.Scenario/TrippinE2ETestCases.cs]: Not for this change: We shall add a StyleCop to make the `this.` optional.

File: test/Microsoft.Restier.EntityFramework.Tests/ModelTests.cs (hunk 2 above)
Comment (karataliu) [inline test/Microsoft.Restier.EntityFramework.Tests/ModelTests.cs]: resolved

File: test/Microsoft.Restier.EntityFramework.Tests/packages.config (hunk 4 above)
Comment (karataliu) [inline test/Microsoft.Restier.EntityFramework.Tests/packages.config]: will merge with that change when check-in

File: test/Microsoft.Restier.EntityFramework.Tests/packages.config (hunk 5 above)
Comment (karataliu) [inline test/Microsoft.Restier.EntityFramework.Tests/packages.config]: added issue https://github.com/OData/RESTier/issues/104

File: test/ODataEndToEndTests/Microsoft.Restier.WebApi.Test.Scenario/TrippinE2ETestCases.cs (hunk 6 above)
Comment (karataliu) [inline test/ODataEndToEndTests/Microsoft.Restier.WebApi.Test.Scenario/TrippinE2ETestCases.cs]: created issue https://github.com/OData/RESTier/issues/105

File: RESTier.sln [hunk 7]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio 2013
-VisualStudioVersion = 12.0.31101.0
+VisualStudioVersion = 12.0.30723.0
```
Comment (lewischeng-ms) [inline RESTier.sln]: do not downgrade the solution version.

File: src/Microsoft.Restier.EntityFramework/Model/ModelProducer.cs [hunk 8]
```diff
@@ -178,13 +177,63 @@ private static void SetComputedAnnotation(EdmModel model, IEdmProperty target)
...
+            {
+                return GetPrimitiveTypeReference(efProperty);
+            }
+            else if (efProperty.IsComplexType)
+            {
+                return GetComplexTypeReference(efProperty, model, elementMap);
+            }
+
+            // TODO GitHubIssue#103 : Choose property error message for unknown type
+            return null;
```
Comment (lewischeng-ms) [inline src/Microsoft.Restier.EntityFramework/Model/ModelProducer.cs]: what about entity type (or navigation property)?

File: test/ODataEndToEndTests/Microsoft.Restier.WebApi.Test.Scenario/ServiceReference/TrippinProxy.tt [hunk 9]
```diff
@@ -11,7 +11,7 @@ public static class Configuration
 	// The use of DataServiceCollection enables entity and property tracking. The value must be set to true or false.
 	public const bool UseDataServiceCollection = true;
 
-	// The namespace of the client code generated. It replaces the original namespace in the metadata document, 
+	// The namespace of the client code generated. It replaces the original namspace in the metadata document,
```
Comment (lewischeng-ms) [inline test/ODataEndToEndTests/Microsoft.Restier.WebApi.Test.Scenario/ServiceReference/TrippinProxy.tt]: why change to a misspelling word.

File: RESTier.sln (hunk 7 above)
Comment (karataliu) [inline RESTier.sln]: undoed

File: test/ODataEndToEndTests/Microsoft.Restier.WebApi.Test.Scenario/ServiceReference/TrippinProxy.tt (hunk 9 above)
Comment (karataliu) [inline test/ODataEndToEndTests/Microsoft.Restier.WebApi.Test.Scenario/ServiceReference/TrippinProxy.tt]: This comes from tt template change, will update with new tt.

File: src/Microsoft.Restier.EntityFramework/Model/ModelProducer.cs (hunk 8 above)
Comment (karataliu) [inline src/Microsoft.Restier.EntityFramework/Model/ModelProducer.cs]: navigation property would be handled by Line 111 with AssociationSets.

**Pull request merged**